File: week1/sexualityAssessment/main.cpp
#include <iostream>
#include <string>

int main(){
    std::string ans1, ans2, ans3, ans4, ans5;
    bool isMale;

    std::cout << "==================== Welcome to the Sexuality Assessment Test ===================" << std::endl;
    std::cout << "Please answer a few of our questions to determine your Sexuality." << std::endl;
    std::cout << "Please answer with yes or no" << std::endl;
    std::cout << "Are you a male?" << std::endl;
    std::cin >> ans1; //Male or Female
    std::cout << "Are you attracted to males?" << std::endl;
    std::cin >> ans2; //Gay or straight
    std::cout << "Are you attracted to females?" << std::endl;
    std::cin >> ans3; //Gay or straight
    std::cout << "Are you attracted to both genders?" << std::endl;
    std::cin >> ans4; //Bisexual
    std::cout << "Are you attracted to anything?" << std::endl;
    std::cin >> ans5; //Pansexual

    if(ans1 == "yes"){
        //Male
        if(ans2 == "no" && ans3 == "yes" && ans4 == "no" && ans5 == "no"){
            //Straight
            std::cout << "You are a straight Male!!" << std::endl;
        }
        else if(ans2 == "yes" && ans3 == "no" && ans4 == "no" && ans5 == "no"){
            //Gay
            std::cout << "You are a Gay!!" << std::endl;
        }
        else if(ans2 == "yes" && ans3 == "yes" && ans4 == "yes" && ans5 == "no"){
            //Bisexual
            std::cout << "You are a Bisexual!!" << std::endl;
        }
        else if(ans2 == "yes" && ans3 == "yes" && ans4 == "yes" && ans5 == "yes"){
            //Pansexual
            std::cout << "You are a Pansexual!!" << std::endl;
        }
        else if(ans2 == "no" && ans3 == "no" && ans4 == "no" && ans5 == "no"){
            //Uncertain
            std::cout << "You are uncertain about your sexuality!!" << std::endl;
        }
    }
    else{
        //Female
        if(ans2 == "yes" && ans3 == "no" && ans4 == "no" && ans5 == "no"){
            //Straight
            std::cout << "You are a straight Female!!" << std::endl;
        }
        else if(ans2 == "no" && ans3 == "yes" && ans4 == "no" && ans5 == "no"){
            //Gay
            std::cout << "You are a Lesbian!!" << std::endl;
        }
        else if(ans2 == "yes" && ans3 == "yes" && ans4 == "yes" && ans5 == "no"){
            //Bisexual
            std::cout << "You are a Bisexual!!" << std::endl;
        }
        else if(ans2 == "yes" && ans3 == "yes" && ans4 == "yes" && ans5 == "yes"){
            //Pansexual
            std::cout << "You are a Pansexual!!" << std::endl;
        }
        else if(ans2 == "no" && ans3 == "no" && ans4 == "no" && ans5 == "no"){
            //Uncertain
            std::cout << "You are uncertain about your sexuality!!" << std::endl;
        }
    }

}

File: week1/sexualityAssessment/main.py
""" 
    Program to determine the sexuality of an individual.
    By Jesse Amarquaye.
    Engineering and Operations Branch, MIS.
    Assignment 1.
    
"""


print("Welcome To The Sexuality Assessment")

try:
    ask_gender = input("Enter your gender. Type 'M' for male and 'F' if you are a Female: " )

    print("Enter 'Y' for Yes or 'N' for No to the following questions to determine your sexuality")

    male_attraction = input("Are you attracted to males? ")
    female_attraction = input("Are you attracted to females? ")
    pan = input("Are you attracted to anything? ")


    # Male Questions
    if ask_gender.lower() == "m":
        gender = "male"

        if (male_attraction.lower() == "y") and (female_attraction.lower() == "y") and (pan.lower() == "y"):
            print("You are Pansexual")

        elif (male_attraction.lower() == "y") and (female_attraction.lower() == "y"):
            print("You are Bisexual")

        elif male_attraction.lower() == "y":
            print("You are Gay")

        elif male_attraction.lower() == "n":
            print("You are straight")

        # MISTAKE
        elif pan.lower() == "y":
            print("You are Pansexual")

        elif pan.lower() == "n":
            if male_attraction.lower() == "y":
                print("Youa are Gay")

            elif female_attraction.lower() == "y":
                print("You are Lesbian")

            elif male_attraction.lower() == "y" and female_attraction.lower() == "y":
                print("You are Bisexual")

            else:
                print("You are Uncertain about your sexuality")


    # Female Questions
    if ask_gender.lower() == "f":
        gender = "female"

        if (male_attraction.lower() == "y") and (female_attraction.lower() == "y") and (pan.lower() == "y"):
            print("You are Pansexual")

        elif (male_attraction.lower() == "y") and (female_attraction.lower() == "y"):
            print("You are Bisexual")

        elif male_attraction.lower() == "y":
            print("You are Straight")

        elif male_attraction.lower() == "n":
            print("You are Lesbian")

        elif pan.lower() == "y":
            print("You are Pansexual")

        elif pan.lower() == "n":
            if male_attraction.lower() == "y":
                print("Youa are Gay")

            elif female_attraction.lower() == "y":
                print("You are Lesbian")

            elif male_attraction.lower() == "y" and female_attraction.lower() == "y":
                print("You are Bisexual")

            else:
                print("You are Uncertain about your sexuality")

except Exception as e:
    print(f"Error {e}")
